cdap-security/src/main/java/io/cdap/cdap/security/authorization/AuthorizerInstantiator.java

```diff
@@ -103,65 +101,70 @@ public AuthorizerInstantiator(CConfiguration cConf, AuthorizationContextFactory
    * authorization is disabled.
    */
   @Override
-  public synchronized Authorizer get() {
-    if (authorizer != null) {
-      return authorizer;
-    }
-
-    if (!authorizationEnabled) {
+  public Authorizer get() {
+    if (!cConf.getBoolean(Constants.Security.Authorization.ENABLED)) {
       LOG.debug("Authorization is disabled. Authorization can be enabled  by setting " +
                   Constants.Security.Authorization.ENABLED + " to true.");
-      authorizer = new NoOpAuthorizer();
-      return authorizer;
+      return NOOP_AUTHORIZER;
     }
-    if (!authenticationEnabled) {
+    if (!cConf.getBoolean(Constants.Security.ENABLED)) {
       LOG.info("Authorization is enabled. However, authentication is disabled. Authorization policies will not be " +
```

Comment:
not related, but should it be at least a warning? It looks like a misconfiguration

Reply:
Sure. It's from the old code.